sqlbrite/src/main/java/com/squareup/sqlbrite/QueryObservable.java

```diff
@@ -41,17 +43,22 @@
       public Subscriber<? super Query> call(final Subscriber<? super List<T>> subscriber) {
         return new Subscriber<Query>(subscriber) {
           @Override public void onNext(Query query) {
-            Cursor cursor = query.run();
-            List<T> items = new ArrayList<>(cursor.getCount());
             try {
-              while (cursor.moveToNext() && !subscriber.isUnsubscribed()) {
-                items.add(mapper.call(cursor));
+              Cursor cursor = query.run();
+              List<T> items = new ArrayList<>(cursor.getCount());
+              try {
+                while (cursor.moveToNext() && !subscriber.isUnsubscribed()) {
+                  items.add(mapper.call(cursor));
+                }
+              } finally {
+                cursor.close();
               }
-            } finally {
-              cursor.close();
-            }
-            if (!subscriber.isUnsubscribed()) {
-              subscriber.onNext(items);
+              if (!subscriber.isUnsubscribed()) {
+                subscriber.onNext(items);
+              }
+            } catch (Throwable e) {
+              Exceptions.throwIfFatal(e);
+              onError(OnErrorThrowable.addValueAsLastCause(e, query));
```

Comment:
I can add a test which will check that `query.toString()` is part of the stacktrace, but it'll require copy-paste from apache commons or I can add it as test dependency, because [retrieving full stacktrace is not so trivial](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/src-html/org/apache/commons/lang/exception/ExceptionUtils.html#line.867).